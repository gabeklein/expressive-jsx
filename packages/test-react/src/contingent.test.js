const test = require("./_adapter");

test("contingent on consequent", `
  () => do {
    if(true){
      div: {
        css: if(".foo-bar")
          background: blue;
      }
    }

    <div>Hello</div>
  }
`);

test("multiple contingent on consequent", `
  () => do {
    if(active){
      css: self: {
        if(".foo")
          color: red;
    
        if(".bar")
          color: red;
      }
    }

    <div>Hello</div>
  }
`);

test("contingent in consequent", `
  () => do {
    if(true){
      div: {
        css: if(".foo-bar")
          background: blue;
      }
    }

    <div>Hello</div>
  }
`);