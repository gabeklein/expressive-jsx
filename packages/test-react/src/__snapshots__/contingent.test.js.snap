// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contingent in consequent 1`] = `
"() => create(\\"div\\", {
  className: !!true && \\"element_3ld\\"
}, create(\\"div\\", {
  className: \\"inner_79m\\"
}, \\"Hello\\"));

css(\`
	/* 5 */
	.element_3ld .inner_79m.foo-bar { background: blue }
\`);"
`;

exports[`contingent in consequent 2`] = `
"() => <div className={!!true && \\"element_3ld\\"}><div className=\\"inner_79m\\">Hello</div></div>;

css(\`
	/* 5 */
	.element_3ld .inner_79m.foo-bar { background: blue }
\`);"
`;

exports[`contingent on consequent 1`] = `
"() => create(\\"div\\", {
  className: !!true && \\"element_3ld\\"
}, create(\\"div\\", {
  className: \\"div_1cq\\"
}, \\"Hello\\"));

css(\`
	/* 5 */
	.element_3ld .div_1cq.foo-bar { background: blue }
\`);"
`;

exports[`contingent on consequent 2`] = `
"() => <div className={!!true && \\"element_3ld\\"}><div className=\\"div_1cq\\">Hello</div></div>;

css(\`
	/* 5 */
	.element_3ld .div_1cq.foo-bar { background: blue }
\`);"
`;

exports[`multiple contingent on consequent 1`] = `
"() => create(\\"div\\", {
  className: !!active && \\"element_3ld\\"
}, create(\\"div\\", {}, \\"Hello\\"));

css(\`
	/* 5 */
	.element_3ld.foo { color: red }
	.element_3ld.bar { color: red }
\`);"
`;

exports[`multiple contingent on consequent 2`] = `
"() => <div className={!!active && \\"element_3ld\\"}><div>Hello</div></div>;

css(\`
	/* 5 */
	.element_3ld.foo { color: red }
	.element_3ld.bar { color: red }
\`);"
`;
